chosen.js

```diff
@@ -90,6 +90,27 @@
           if (attr.ngOptions && ngModel) {
             match = attr.ngOptions.match(NG_OPTIONS_REGEXP);
             valuesExpr = match[7];
+            var options = scope.$eval(valuesExpr);
+            // Adding title in case of array of options and object data source
+            if (options && typeof options === 'object') {
```

Comment:
typeof is not that appropriate when dealing with objects or arrays.

Reply:
As discussed with you, typeof can be used here